adal/devApps/XFormsApp/SecondPage.cs

```diff
@@ -202,7 +208,54 @@ void LogCallback(LogLevel level, string message, bool containsPii)
                     this.result.Text += "Logs : " + DrainLogs();
                 });
             }
+        }
 
+        private async void AcquireTokenWithBrokerButton_Clicked(object sender, EventArgs e)
+        {
+            this.result.Text = string.Empty;
+            AuthenticationContext ctx = new AuthenticationContext("https://login.microsoftonline.com/common");
+            string output = string.Empty;
+            string accessToken = String.Empty;
+            this.testResult.Text = "Success:";
+            
+            try
+            {
+                AuthenticationResult result =
+                    await
+                        ctx.AcquireTokenAsync("https://graph.microsoft.com", ClientId,
+                            new Uri(DeterminePlatformForRedirectUri()),
+                            Parameters).ConfigureAwait(false);
+                output = "Signed in User - " + result.UserInfo.DisplayableId;
+                accessToken = result.AccessToken;
+            }
+            catch (Exception exc)
+            {
+                output = exc.Message;
+            }
+            finally
+            {
+                Device.BeginInvokeOnMainThread(() =>
+                {
+                    this.testResult.Text = string.IsNullOrWhiteSpace(accessToken) ? "Success: False" : "Success: True";
+                    this.result.Text += "Result : " + output;
+
+                    this.result.Text += "Logs : " + DrainLogs();
+                });
+            }
+        }
+
+        private string DeterminePlatformForRedirectUri()
+        {
+            switch (Device.RuntimePlatform)
+            {
+                case Device.iOS:
+                    RedirectURI = IOSBrokerRedirectURI;
+                    break;
+                case Device.Android:
+                    RedirectURI = AndroidBrokerRedirectURI;
+                    break;
```

Comment:
Nit: I like to add a default: throw new NotImplementedException just in case ... fail fast if another runtime platform is used.

Reply:
agree...sounds good. thanks

Reply:
done.